compat/src/index.d.ts

```diff
@@ -0,0 +1,50 @@
+import * as _hooks from '../../hooks';
+import * as preact from '../../src';
+import { ForwardFn } from './internal';
+
+export = compat;
+export as namespace compat;
+declare namespace compat {
```

Comment:
do we need a namespace here?

Reply:
Likely not, I was hoping to try and facilitate both exporting everything as default, and individually using this, but it doesn't seem to work as I had anticipated.

Reply:
I removed the namespace. Still thinking it would be nice to support a default export as well as named, but nothing I have seem really shows a nice way of doing this.

Reply:
There is always the option to use an `object` for the default export:

```ts
export const foo: number;
export function bar() {}

export default {
  foo,
  bar,
}
```

Reply:
Resolved in the most recent commit. Thanks!